mlflow/sagemaker/container/__init__.py

```diff
@@ -87,35 +99,52 @@ def _serve():
            "mlflow.sagemaker.container.scoring_server.wsgi:app").format(nworkers=cpu_count)
     bash_cmds.append(cmd)
     gunicorn = Popen(["/bin/bash", "-c", "; ".join(bash_cmds)])
-    signal.signal(signal.SIGTERM, lambda a, b: _sigterm_handler(nginx.pid, gunicorn.pid))
+    signal.signal(signal.SIGTERM, lambda a, b: _sigterm_handler(pids=[nginx.pid, gunicorn.pid]))
     # If either subprocess exits, so do we.
-    pids = set([nginx.pid, gunicorn.pid])
-    while True:
-        pid, _ = os.wait()
-        if pid in pids:
-            break
-    _sigterm_handler(nginx.pid, gunicorn.pid)
+    awaited_pids = _await_subprocess_exit_any(procs=[nginx, gunicorn])
+    _sigterm_handler(awaited_pids)
```

Comment:
Maybe move the _sigterm handler inside _await_... function?

Reply:
True but you could still call _sigterm_handler inside await... ? It's really a nit, not important.

Reply:
Ah, I see. I think the `_await...` function may be useful on its own as a means of waiting without terminating.